docs/lava-blockchain/join-mainnet-manual-cosmovisor.md

```diff
@@ -0,0 +1,218 @@
+        ### Packages installations
+        sudo apt update # in case of permissions error, try running with sudo
+        sudo apt install -y unzip logrotate git jq sed wget curl coreutils systemd
+        # Create the temp dir for the installation
+        temp_folder=$(mktemp -d) && cd $temp_folder
+        ```
+        
+    - Go installation
+        
+        ```bash
+        ### Configurations
+        go_package_url="https://go.dev/dl/go1.20.5.linux-amd64.tar.gz"
+        go_package_file_name=${go_package_url##*\/}
+        # Download GO
+        wget -q $go_package_url
+        # Unpack the GO installation file
+        sudo tar -C /usr/local -xzf $go_package_file_name
+        # Environment adjustments
+        echo "export PATH=\$PATH:/usr/local/go/bin" >>~/.profile
+        echo "export PATH=\$PATH:\$(go env GOPATH)/bin" >>~/.profile
+        source ~/.profile
+        ```
+        
+    - Installation verifications
+        
+        
+        1. You can verify the installed go version by running: `go version`
+        
+        2. The command `go env GOPATH` should include `$HOME/go`
+        If not, then, `export GOPATH=$HOME/go`
+        
+        3. PATH should include `$HOME/go/bin`
+        To verify PATH, run `echo $PATH`
+        
+
+## 1. Set up a local node
+
+### Download app configurations
+
+- Download setup configuration
+    
+    Download the configuration files needed for the installation
+    
+    ```bash
+    # Download the installation setup configuration
+    git clone https://github.com/lavanet/lava-config.git
+    cd lava-config/mainnet
+    # Read the configuration from the file
+    # Note: you can take a look at the config file and verify configurations
+    source setup_config/setup_config.sh
+    ```
+    
+- Set app configurations
+        
+    Copy lavad default config files to config Lava config folder
+    
+    ```bash
+    echo "Lava config file path: $lava_config_folder"
+    mkdir -p $lavad_home_folder
+    mkdir -p $lava_config_folder
+    cp default_lavad_config_files/* $lava_config_folder
+    ```
+    
+
+### Set the genesis file
+
+- Set the genesis file in the configuration folder
+    
+    ```bash
+    # Copy the genesis.json file to the Lava config folder
+    cp genesis_json/genesis.json $lava_config_folder/genesis.json
+    ```
+
+## 2. Join the Lava Mainnet
+
+The following sections will describe how to install Cosmovisor for automating the upgrades process.
+
+
+### Set up Cosmovisor {#cosmovisor}
+
+- Set up cosmovisor to ensure any future upgrades happen flawlessly. To install Cosmovisor:
+    
+    ```bash
+    go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@v1.0.0
+    # Create the Cosmovisor folder and copy config files to it
+    mkdir -p $lavad_home_folder/cosmovisor/genesis/bin/
+    # Download the genesis binary
+    wget -O  $lavad_home_folder/cosmovisor/genesis/bin/lavad "https://github.com/lavanet/lava/releases/download/v2.2.0/lavad-v2.2.0-linux-amd64"
```

Comment:
this version will work only if we use snapshot or statesync.
otherwise it is an archive node and needs the first version released for mainnet

Reply:
fixed to "https://github.com/lavanet/lava/releases/download/v0.33.0/lavad-v0.33.0-linux-amd64"